website/blog/2024-11-27-test-smarter-part-2.md

```diff
@@ -0,0 +1,123 @@
+---
+title: "Test smarter: Where should tests go in your pipeline?"
+description: "Testing your data should drive action, not accumulate alerts. We take our testing framework developed in our last post and make recommendations for where tests ought to go at each transformation stage."
+slug: test-smarter-where-tests-should-go
+
+authors: [faith_mckenna, jerrie_kumalah_kenney]
+
+tags: [analytics craft]
+hide_table_of_contents: false
+
+date: 2024-12-09
+is_featured: true
+---
+
+👋 Greetings, dbt’ers! It’s Faith & Jerrie, back again to offer tactical advice on *where* to put tests in your pipeline.
+
+In [our first post](https://docs.getdbt.com/blog/test-smarter-not-harder) on refining testing best practices, we developed a prioritized list of data quality concerns. We also documented first steps for debugging each concern. This post will guide you on where specific tests should go in your data pipeline.
+
+*Note that we are constructing this guidance based on how we [structure data at dbt Labs.](https://docs.getdbt.com/best-practices/how-we-structure/1-guide-overview#guide-structure-overview)* You may use a different modeling approach—that’s okay! Translate our guidance to your data’s shape, and let us know in the comments section what modifications you made. 
+
```

Comment:
@faithebear @matthewshaver not on my computer but trying this to see if it works since we need to have the truncate

```suggestion

<--! truncate -->

```

Reply:
Nevermind lol, thanks Matt!

Reply:
I'd suggest moving the truncate block to after the list below (i.e. before the Where should tests go h2), so that the key takeaways are included in the preview, not just the disclaimers.

Reply:
did that in most recent commit